django/middleware/csrf.py

```diff
@@ -155,9 +158,29 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 if referer is None:
                     return self._reject(request, REASON_NO_REFERER)
 
-                # Note that request.get_host() includes the port.
-                good_referer = 'https://%s/' % request.get_host()
-                if not same_origin(referer, good_referer):
+                referer = urlparse(referer)
+
+                # Make sure we have a valid URL for Referer
+                if '' in (referer.scheme, referer.netloc):
+                    return self._reject(request, REASON_MALFORMED_REFERER)
```

Comment:
The `MALFORMED_REFERER` branch was already covered by `test_https_malformed_referer`. Added a test to cover the `INSECURE_REFERER` branch